plaso/engine/artifacts_filter_file.py

```diff
@@ -0,0 +1,272 @@
+# -*- coding: utf-8 -*-
+"""Filter file."""
+
+from __future__ import unicode_literals
+
+import logging
+import re
+
+from artifacts import definitions as artifact_types
+from artifacts import errors as artifacts_errors
+from artifacts import reader as artifacts_reader
+from artifacts import registry as artifacts_registry
+
+from dfvfs.helpers import file_system_searcher
+from dfwinreg import registry_searcher
+
+
+from plaso.lib import py2to3
+from plaso.lib import errors
+
+
+ARTIFACTS_FILTER_FILE = 'ARTIFACTS_FILTER_FILE'
+COMPATIBLE_DFWINREG_KEYS = [r'HKEY_LOCAL_MACHINE',
+                            r'HKEY_LOCAL_MACHINE\SYSTEM',
+                            r'HKEY_LOCAL_MACHINE\SOFTWARE',
+                            r'HKEY_LOCAL_MACHINE\SAM',
+                            r'HKEY_LOCAL_MACHINE\SECURITY']
+RECURSIVE_GLOB_LIMIT = 10
+
+
+class ArtifactsFilterFile(object):
+  """Forensic artifacts file.
+
+  A forensic artifacts file contains one or more forensic artifacts filters.
+
+  Forensic artifacts are defined in:
+  https://github.com/ForensicArtifacts/artifacts/blob/master/docs/Artifacts%20definition%20format%20and%20style%20guide.asciidoc
+  """
+
+  def __init__(self, path, knowledge_base):
+    """Initializes a filter file.
+
+    Args:
+      path (str): path to a file that contains one or more forensic artifacts.
+      knowledge_base (KnowledgeBase): Knowledge base for Log2Timeline.
+    """
+    super(ArtifactsFilterFile, self).__init__()
+    self._path = path
+    self._knowledge_base = knowledge_base
+
+  def BuildFindSpecs(self, environment_variables=None):
+    """Build find specification from a forensic artifacts file.
+
+    Args:
+      environment_variables (Optional[list[EnvironmentVariableArtifact]]):
+          environment variables.
+    """
+    find_specs = {}
+    path_attributes = self._BuildPathAttributes(environment_variables)
+    artifact_registry = artifacts_registry.ArtifactDefinitionsRegistry()
+    artifact_reader = artifacts_reader.YamlArtifactsReader()
+
+    try:
+      artifact_registry.ReadFromFile(artifact_reader, self._path)
+
+    except (KeyError, artifacts_errors.FormatError) as exception:
+      raise errors.BadConfigOption((
+          'Unable to read artifact definitions from: {0:s} with '
+          'error: ''{1!s}').format(self._path, exception))
```

Comment:
why the double ' before {1!s} ?

Reply:
Typo that pycharm completed for me, removed.